trie/trieChangesBatch/trieChangesBatch.go

```diff
@@ -107,8 +108,9 @@ func (t *trieChangesBatch) IsInterfaceNil() bool {
 
 func getSortedData(dataMap map[string]core.TrieData) []core.TrieData {
 	data := make([]core.TrieData, 0, len(dataMap))
-	for k := range dataMap {
-		data = append(data, dataMap[k])
+	for _, trieData := range dataMap {
+		trieData.Key = keyBuilder.KeyBytesToHex(trieData.Key)
```

Comment:
why would we need a hex encode here? 
Can't we delete hex encode and decode from everywhere ?

Reply:
We need the trie keys to be hex encoded because the branchNodes can have 16 children, and based on the key we build the trie. If we were to use a binary trie, we would have converted the keys to binary.